### DISCUSSION TITLE
Thermostat showing as sensors but not as climate objects

### DISCUSSION THREAD

**Discussion by piottolorenzo** [gicamm/homeassistant-comelit#60] [Q&A]
Hi, 

I have successfully connected home assistant to the comelit hub. 
I have one issue with all the comelit thermostat as they show up as sensors (can read temperature) but not as climate objects. Therefore I cannot change temperature or turn them on/off. The only one that I can set is the one embedded in the monitor maxi (#6813W)
Thermostats are #20046709 (https://pro.comelitgroup.com/product/20046709)

Any of you has experience with this?

Thank you in advance

**Comment by gicamm** [comment]
Hi, what type of thermostat do you have?
Do you also have another thermostat?

**Reply by piottolorenzo** [reply]
Hi thanks for the prompt reply. 

I have 5 thermostats in total. 
1 is embedded in the comelit maxi manager (the display) and this shows up correctly in home assistant 
The other 4 are comelit 20046709. They appear as sensors in home assistant and I correctly see the temperature reading, but they have no climate object so I cannot control them.
(https://pro.comelitgroup.com/product/20046709)

**Reply by gicamm** [reply]
Unfortunately, I do not have this thermostat, so it is not easy for me to integrate it. Can you write code? You could contribute.

**Comment by piottolorenzo** [comment]
I'm an electronic engineer so I have background on programming, not that
much in python though.
I could give it a try if you could somehow advise me in the process! Al
least to get the full picture and understand the flow

**Comment by gicamm** [comment]
You're welcome :D
The process is quite simple.
The integration extracts data from the json that the MQTT server (the Comelit HUB) sends. So you need to look at that json and figure out if the hub is exporting more than just what the integration classifies as a sensor.
In terms of code, you need to look at 
[https://github.com/gicamm/homeassistant-comelit/blob/master/custom_components/comelit/hub.py:update_entities](https://github.com/gicamm/homeassistant-comelit/blob/master/custom_components/comelit/hub.py:update_entities)

You can then subscribe your own client to the MQTT Publisher and view the data.
[https://github.com/gicamm/homeassistant-comelit/blob/master/test/mqtt.txt](https://github.com/gicamm/homeassistant-comelit/blob/master/test/mqtt.txt)

**Reply by piottolorenzo** [reply]
Thanks a lot for the hints.. took me some time to get into it but I was able to find 2 issues. 
First it looks like they changed the sub_type for climate object from 16 to 12, also for the one in the maxi manager. I recently updated the hub firmware to 1.7 and in fact that thermostat, which worked in the past, was not working anymore. 
Second issue is that the other thermostats don't have a humidity sensors, so the execution was crashing. To make it work I manually removed everithing realted to humidity.
BTW looks to me they also changed the humidity infos for the maxi manager itself as I see them only with req_type=15 and not in req_type=0

I would say debug is done. I can try to implement something to fix/overcome the issue but this will take me more time :)

**Reply by gicamm** [reply]
Take your time. No rush! :D
